### DISCUSSION TITLE
Potential Bug? Exporting from Marp-VScode defaults to HTML when PDF is selected

### DISCUSSION THREAD

**Discussion by zhangchuck** [marp-team/marp#148] [Q&A] [answered]
When I export from Marp-VSCode, I get the dialog below:

![image](https://user-images.githubusercontent.com/4514597/130192701-0d063f79-bf3e-4aa7-acab-055e8cca333c.png)

- If I enter a filename without extension, VS Code will export an HTML slide deck (even though "PDF slide deck" is selected in the bottom right drop-down box.
- If I enter a filename with the `.pdf` extension, VS Code exports a PDF

Does this happen for anyone else? Is this a bug?

**Environment:** Fedora 34 / VS Code 1.59 / Marp for VSCode 1.2.0

**Comment by yhatt** [comment]
https://github.com/marp-team/marp-vscode/issues/247#issuecomment-855996933

> The export format actually relies the extension in the filename specified in save dialog (e.g. `slide.html`, `slide.pdf`) and _the content of slide deck does not matter_. If not detected the extension from filename, Marp will try to export into HTML with regardless of setting.
> 
> Try to set a extension `.pdf` or `.pptx` explicitly.
> 
> > Normally a corresponded extension with selected type will add automatically by VS Code, but some Linux distributions have reported won't add. Potentially related: [#83](https://github.com/marp-team/marp-vscode/issues/83)

It's not a bug of the extension. VS Code has no API to know selected type so we have no choice but to depend to the extension.

**Reply by zhangchuck** [reply]
Thanks, I missed this when I was searching the old issues.

Looks like it must be an issue from either VS Code or Cinnamon (my window manager).

**Comment by yhatt** [comment, marked as answer]
We think this confusion was caused from a default filename without extension.

I have refactored to contain a file extension into the default filename for an export dialog. This change has already shipped since v1.3.0.

I had concerned about double file extensions in other platforms (`xxxx.pdf.pdf`), but it seems to have been all in my mind: we have not received any reports about that.

**Reply by zhangchuck** [reply]
Yeah, this works pretty well. It's now explicit what is being produced. Previously, mildly confusing because the drop-down would highlight PDF but you would get a HTML deck if you didn't explicitly add `.pdf`. Thanks for the change!
